homeassistant/components/remote/xiaomi_miio.py

```diff
@@ -0,0 +1,254 @@
+        start_time = utcnow()
+        while (utcnow() - start_time) < timedelta(seconds=timeout):
+            message = yield from hass.async_add_job(
+                device.read, slot)
+            _LOGGER.debug("Message recieved from device: '%s'", message)
+            if 'code' in message and message['code']:
+                log_msg = "Received command is: {}".\
+                          format(message['code'])
+                _LOGGER.info(log_msg)
+                hass.components.persistent_notification.async_create(
+                    log_msg, title='Xiaomi Miio Remote')
+                return
+            if ('error' in message and
+                    message['error']['message'] == "learn timeout"):
+                yield from hass.async_add_job(device.learn, slot)
+            yield from asyncio.sleep(1, loop=hass.loop)
+        _LOGGER.error("Timeout. No infrared command captured")
+        hass.components.persistent_notification.async_create(
+            "Timeout. No infrared command captured",
+            title='Xiaomi Miio Remote')
+
+    hass.services.async_register(DOMAIN, SERVICE_LEARN, async_service_handler,
+                                 schema=LEARN_COMMAND_SCHEMA)
+
+
+class XiaomiMiioRemote(Entity):
+    """Representation of a Xiaomi Miio Remote device."""
+
+    def __init__(self, friendly_name, device,
+                 slot, timeout, hidden, commands):
+        """Initialize the remote."""
+        self._name = friendly_name
+        self._device = device
+        self._is_hidden = hidden
+        self._slot = slot
+        self._timeout = timeout
+        self._state = False
+        self._commands = commands
+
+    @property
+    def name(self):
+        """Return the name of the remote."""
+        return self._name
+
+    @property
+    def device(self):
+        """Return the remote object."""
+        return self._device
+
+    @property
+    def hidden(self):
+        """Return if we should hide entity."""
+        return self._is_hidden
+
+    @property
+    def slot(self):
+        """Return the slot to save learned command."""
+        return self._slot
+
+    @property
+    def timeout(self):
+        """Return the timeout for learning command."""
+        return self._timeout
+
+    @property
+    def is_on(self):
+        """Return False if device is unreachable, else True."""
+        from miio import DeviceException
+        try:
+            self.device.info()
+            return True
+        except DeviceException:
+            return False
+
+    @property
+    def should_poll(self):
+        """We should be polled for device up state."""
+        return True
+
+    @property
+    def device_state_attributes(self):
+        """Hide remote by default."""
+        if self._is_hidden:
+            return {'hidden': 'true'}
+        else:
+            return
+
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
+        """Turn the device on."""
```

Comment:
I'd add `pass` after the comment to make it explicit that these are not supposed to do anything. Or even better, implement them properly / add a logger warning when someone tries to call them.

At the moment someone converting from another remote platform to this will struggle to see why their commands are not working as expected.

Reply:
I'm going to add an error message, in case people try to turn it off or on, as I really do feel that a "switch" for this type of device is bad UX. There are multiple problems with having a "switch" for an IR "bridge":
1. It is always going to be optimistic, we have no way of knowing the real state of the device we are controlling.
2. Most devices use the same IR code for on/off, why would we even have a switch for those cases?

If at all possible I would rather the bridge device just reported the state (Online/Offline) with no possibility of turning it on and off, I feel like the would be a much better UX for this kind of device, but I don't really have any idea about how to implement that, so I'm going to throw an error, if people try turning it off or on.